sdk/python/tests/expediagroup/test_milvus_online_store.py

```diff
@@ -1,37 +1,293 @@
+            alias=ALIAS, host=HOST, username="abc", password="cde"
+        ),
+        offline_store=FileOfflineStoreConfig(),
+        entity_key_serialization_version=2,
+    )
 
-def test_milvus_start_stop():
-    # this is just an example how to start / stop Milvus. Once a real test is implemented this test can be deleted
+
+@pytest.fixture
+def milvus_online_store():
+    return MilvusOnlineStore()
+
+
+@pytest.fixture(scope="session")
+def milvus_online_setup():
+    # Creating an online store through embedded Milvus for all tests in the class
     online_store_creator = MilvusOnlineStoreCreator("milvus")
     online_store_creator.create_online_store()
 
-    # access through a stub
-    milvus = Milvus(online_store_creator.host, 19530)
-
-    # Create collection demo_collection if it doesn't exist.
-    collection_name = "example_collection_"
-
-    ok = milvus.has_collection(collection_name)
-    if not ok:
-        fields = {
-            "fields": [
-                {"name": "key", "type": DataType.INT64, "is_primary": True},
-                {
-                    "name": "vector",
-                    "type": DataType.FLOAT_VECTOR,
-                    "params": {"dim": 32},
-                },
+    yield online_store_creator
+
+    # Tearing down the Milvus instance after all tests in the class
+    online_store_creator.teardown()
+
+
+class TestMilvusConnectionManager:
+    def test_connection_manager(self, repo_config, caplog, milvus_online_setup, mocker):
+
+        mocker.patch("pymilvus.connections.connect")
+        with MilvusConnectionManager(repo_config.online_store):
+            assert (
+                f"Connecting to Milvus with alias {repo_config.online_store.alias} and host {repo_config.online_store.host} and default port {repo_config.online_store.port}."
+                in caplog.text
+            )
+
+        connections.connect.assert_called_once_with(
+            host=repo_config.online_store.host,
+            username=repo_config.online_store.username,
+            password=repo_config.online_store.password,
+            use_secure=True,
+        )
+
+    def test_context_manager_exit(
+        self, repo_config, caplog, milvus_online_setup, mocker
+    ):
+        # Create a mock for connections.disconnect
+        mock_disconnect = mocker.patch("pymilvus.connections.disconnect")
+
+        # Create a mock logger to capture log calls
+        mock_logger = mocker.patch(
+            "feast.expediagroup.vectordb.milvus_online_store.logger", autospec=True
+        )
+
+        with MilvusConnectionManager(repo_config.online_store):
+            print("Doing something")
+
+        # Assert that connections.disconnect was called with the expected argument
+        mock_disconnect.assert_called_once_with(repo_config.online_store.alias)
+
+        # Assert that logger.info was called with expected messages
+        expected_log_calls = [
+            mocker.call("Closing the connection to Milvus"),
```

Comment:
verification through logs is not a good practice in unit tests. Look into methods of "connections" like `get_connection_addr()` https://github.com/milvus-io/pymilvus/blob/master/pymilvus/orm/connections.py#L396 to use for the assert statements